multiversx_sdk_core/transaction_builders/delegation_factory.py

```diff
@@ -0,0 +1,430 @@
+from typing import Any, List, Optional, Protocol, Sequence, Tuple
+
+from multiversx_sdk_core import TransactionPayload
+from multiversx_sdk_core.constants import (ARGS_SEPARATOR,
+                                           TRANSACTION_OPTIONS_DEFAULT,
+                                           TRANSACTION_VERSION_DEFAULT)
+from multiversx_sdk_core.errors import ErrListsLengthDoNotMatch
+from multiversx_sdk_core.interfaces import (IAddress, IChainID, IGasLimit,
+                                            IGasPrice, INonce, ISignature,
+                                            ITransactionPayload,
+                                            ITransactionValue,
+                                            IValidatorPublicKey)
+from multiversx_sdk_core.serializer import arg_to_string
+from multiversx_sdk_core.transaction import Transaction
+
+
+class IConfig(Protocol):
+    chain_id: IChainID
+    min_gas_price: IGasPrice
+    min_gas_limit: IGasLimit
+    gas_limit_per_byte: IGasLimit
+
+
+class MetaChainSystemSCsCost:
+    STAKE = 5000000
+    UNSTAKE = 5000000
+    UNBOND = 5000000
+    CLAIM = 5000000
+    GET = 5000000
+    CHANGE_REWARD_ADDRESS = 5000000
+    CHANGE_VALIDATOR_KEYS = 5000000
+    UNJAIL = 5000000
+    DELEGATION_MANAGER_OPS = 50000000
+    DELEGATION_OPS = 1000000
+    UNSTAKE_TOKENS = 5000000
+    UNBOND_TOKENS = 5000000
+
+
+ADDITIONAL_GAS_LIMIT_PER_NODE = 6000000
+ADDITIONAL_GAS_FOR_OPERATIONS = 10000000
+
+
+class DelegationFactory:
+    def __init__(self, config: IConfig) -> None:
+        self.config = config
+
+    def create_new_delegation_contract(self,
+                                       sender: IAddress,
+                                       receiver: IAddress,
+                                       total_delegation_cap: int,
+                                       service_fee: int,
+                                       value: ITransactionValue,
+                                       transaction_nonce: Optional[INonce] = None,
+                                       gas_price: Optional[IGasPrice] = None,
+                                       gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        data = self._prepare_data_for_create_new_delegation_contract(total_delegation_cap, service_fee)
```

Comment:
Although not mandatory, it would be nice to have the same "inner" design as in:

 - https://github.com/multiversx/mx-sdk-js-core/blob/main/src/tokenOperations/tokenOperationsFactory.ts
 - https://github.com/multiversx/mx-sdk-py-core/blob/token-operations/multiversx_sdk_core/token_operations/token_operations_factory.py

That is, `_prepare_data_for_create_new_delegation_contract` can be merged directly into `create_new_delegation_contract`, and we can use the name `parts` instead of `args_list` (after all, `function` is not an argument per se).

Reply:
Done!